lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1167,14 +1198,24 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
 
         const scope = scopes.join(" ").toLowerCase();
         const userObject = user ? user : this.getUser();
+        const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
         //if user is not currently logged in and no login_hint/sid is passed as an extraQueryParamater
-          if (!userObject && !(extraQueryParameters &&  ((extraQueryParameters.indexOf(Constants.login_hint) !== -1 ||  extraQueryParameters.indexOf(Constants.sid) !== -1 )))) {
+          if (!userObject && this.checkSSO(extraQueryParameters) && Utils.isEmpty(adalIdToken) ) {
               this._logger.info('User login is required');
               reject(ErrorCodes.userLoginError + Constants.resourceDelimeter + ErrorDescription.userLoginError);
               return;
           }
+          //if user didn't passes the login_hint and adal's idtoken is present and no userobject, use the login_hint from adal's idToken
+          else if(!userObject && !Utils.isEmpty(adalIdToken))
```

Comment:
wouldn't the method that calls ATS already take care of this (parsing id token and putting upn to extraQueryparams)?